### PR TITLE
min 7.0 - min 7.7

### LINKED ISSUE #7: 7.0 Admin Portal
Admin Portal for adding, removing business

### LINKED ISSUE #74: 7.1 Create Migration for Admin

### LINKED ISSUE #77: 7.2 Create Admin login functionality

### LINKED ISSUE #79: 7.3 Create Admin Portal

### LINKED ISSUE #80: 7.4 Make Admin/User Persistent

### LINKED ISSUE #81: 7.5 Logout Admin/Users

### PR DESCRIPTION
**Pull request by deontae-smith**
Closes #7 
Closes #81 
Closes #80 
Closes #77 
Closes #74 
Closes #79 

# Images 
Screenshot 2024-02-23 at 11 44 21 PM

Screenshot 2024-02-23 at 11 41 15 PM

Screenshot 2024-02-24 at 1 18 13 AM

Screenshot 2024-02-22 at 4 21 44 PM

### CODE REVIEW

File: apps/backend/node.ts [hunk 1]
```diff
@@ -55,11 +62,29 @@ export async function createMinBusinessServer(app: any) {
...
 node.get('/', async (req, res) => {
   return res.status(200).send({ message: 'Port opended, see /graphql' });
 });
 
+node.post('/admin/actions/suspend', async (req, res) => {
+  if (!req.body.id) {
+    return res.status(400).send({ error: 'No data provided' });
+  }
+  try {
+    await handleSuspendUser({
```
Comment (deontae-smith) [inline apps/backend/node.ts:77]: nit: HandleSuspendBusiness

File: apps/backend/src/controllers/suspendBusiness.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+const ensureSqlExists = () => {
+  if (fs.existsSync('../migrations/sql')) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+export type Suspend = { id: string; adminName: string };
```
Comment (deontae-smith) [inline apps/backend/src/controllers/suspendBusiness.ts:14]: move to actions-ison

File: apps/backend/src/controllers/suspendBusiness.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+function uploadMigration(contents: string, name: string) {
+  const __filename = fileURLToPath(import.meta.url);
+  const __dirname = dirname(__filename);
+
+  const directory = path.join(__dirname, '..', '..', '..', 'migrations', 'sql');
+  const where = path.join(directory, name);
+
+  fs.writeFile(where, contents, (err) => {
+    if (err) {
+      console.error('Error with admin action', err);
```
Comment (deontae-smith) [inline apps/backend/src/controllers/suspendBusiness.ts:49]: should prop throw an error here

File: apps/backend/src/controllers/suspendBusiness.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+};
+
+async function deleteFile(filePath: string) {
+  try {
+    await fs.promises.unlink(filePath);
+    console.log(`File ${filePath} deleted successfully.`);
+  } catch (err) {
+    console.error(`Error deleting file ${filePath}:`, err);
+  }
+}
```
Comment (deontae-smith) [inline apps/backend/src/controllers/suspendBusiness.ts:75]: can be move to action-iso

File: apps/backend/src/controllers/suspendBusiness.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+const runMigration = async (migration: string) => {
```
Comment (deontae-smith) [inline apps/backend/src/controllers/suspendBusiness.ts:77]: move to actions-iso

**Review by doronr18 — APPROVED** [review]

**Pull request merged**